lib/RemoteAST/RemoteAST.cpp

```diff
@@ -1162,6 +1165,35 @@ class RemoteASTContextConcreteImpl final : public RemoteASTContextImpl {
     if (result) return RemoteAddress(*result);
     return getFailure<RemoteAddress>();
   }
+
+  //  OpaqueExistentialContainer layout.
+  //  struct OpaqueExistentialContainer {
+  //  void *fixedSizeBuffer[3];
+  //  Metadata *type;
+  //  WitnessTable *witnessTables[NUM_WITNESS_TABLES];
+  // };
+  Result<std::pair<Type, uint64_t>>
+  getDynamicTypeAndAddressOpaqueExistential(RemoteAddress object) {
+    uint64_t offset = 3 * sizeof(void *);
+    auto MetadataAddressAddress =
+        RemoteAddress(object.getAddressData() + offset);
+    int64_t metadataAddress;
+    if (!readWordOffset(MetadataAddressAddress, &metadataAddress))
+      return getFailure<std::pair<Type, uint64_t>>();
+    auto result = Reader.readTypeFromMetadata(metadataAddress);
+    if (!result)
+      return getFailure<std::pair<Type, uint64_t>>();
+    // XXX: FIXME: offset.
+    return std::make_pair<Type, uint64_t>(std::move(result), 0);
```

Comment:
For figuring out the value address, check out how `OpaqueExistentialContainer::projectValue()` is implemented.